Feat: `fixTableCellAlign` option

jorroll opened the pull request:
<!--
  Please check the needed checkboxes ([ ] -> [x]). Leave the
  comments as they are, they won’t show on GitHub.
  We are excited about pull requests, but please try to limit the scope, provide
  a general description of the changes, and remember, it’s up to you to convince
  us to land it.
-->

### Initial checklist

* [x] I read the support docs <!-- https://github.com/rehypejs/.github/blob/main/support.md -->
* [x] I read the contributing guide <!-- https://github.com/rehypejs/.github/blob/main/contributing.md -->
* [x] I agree to follow the code of conduct <!-- https://github.com/rehypejs/.github/blob/main/code-of-conduct.md -->
* [x] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=user%3Arehypejs&type=Issues -->
* [x] If applicable, I’ve added docs and tests

### Description of changes

Adds a new plugin option, ~~`mapDepricatedTableCellAttrsToInlineCss`~~`fixTableCellAlign`, which controls whether table-related attributes are mapped to inline CSS styles. Defaults to `true`.

Closes #47

_Notes:_ 
1. I created this using Github's online vscode editor (i.e. press `.` when viewing a repo) so I'm not sure if this conforms to any required lint/formatting rules.
2. My own preference is to default to clear, verbose names, but I won't be offended if you'd prefer to rename this option to something else.

<!--do not edit: pr-->

jorroll commented:
Looks like a ts-ignore which I removed was necessary after all. I'll fix and add some tests...

jorroll commented:
Ready for review

wooorm commented on lib/index.js:
```diff
@@ -19,6 +19,11 @@
  *   You should typically pass `React.Fragment`.
  * @property {string|undefined} [prefix='h-']
  *   React key prefix.
+ * @property {boolean|undefined} [mapDepricatedTableCellAttrsToInlineCss=true]
```
```suggestion
 * @property {boolean|undefined} [fixTableCellAlign=true]
```

The current name is a bit long. What about this?
It’s just one idea, maybe you have more/better ones!

wooorm commented on lib/index.js:
```diff
@@ -19,6 +19,11 @@
  * @property {string|undefined} [prefix='h-']
  *   React key prefix.
+ * @property {boolean|undefined} [mapDepricatedTableCellAttrsToInlineCss=true]
+ *   Default: true. Map deprecated table-related attributes to css styles as
+ *   recommended by Mozilla Developer Network docs. E.g.
+ *   `align="center"` to `style="text-align: center;"`. May result
+ *   in rendering differences.
```
Some recommendations:

```suggestion
 *   Fix obsolete align attributes on table cells by turning them into inline
 *   styles.
 *   Keep it on when working with markdown, turn it off when working with
 *   markup for emails.
```

Feel free to iterate on this.

* Uses [semantic line breaks](https://sembr.org)
* `default: true` is already added with the `[xxx=true]` notation
* the goal of this feature is to solve obsolete attributes, not just because MDN says it too
* adds a recommendation on when to have it on/off

jorroll replied:
I accepted this change with the exception that I kept the `Default: true` comment at the end. When typescript processes this comment and creates the `SharedOptions` interface, it keeps the JSDoc comments but doesn't keep any reference to `[fixTableCellAlign=true]` (so, when inspecting the type in the editor, you don't know that it has a default value). By also referencing the default value in the comment, we ensure that the generated type retains this information.

For example, here's what I see when I inspect the `prefix` option in VS Code. There is no reference to a default value in the editor tooltip and there also isn't a reference to it in the generated typescript type.

<img width="984" alt="Screenshot 2023-04-13 at 11 08 38 AM" src="https://user-images.githubusercontent.com/12599934/231836729-69b50bdc-b242-40d4-b89d-85ebe93440e3.png">

wooorm commented on lib/index.js:
```diff
@@ -53,11 +58,18 @@ export default function rehypeReact(options) {
 
   Object.assign(this, {Compiler: compiler})
 
+  const mapDepricatedTableCellAttrsToInlineCss =
```
Move this next to the other assignment (`const createElement`)

wooorm commented on readme.md:
```diff
@@ -174,6 +174,13 @@ React key prefix (`string`, default: `'h-'`).
 
+###### `options.mapDepricatedTableCellAttrsToInlineCss`
+
+Map deprecated table-related attributes to css styles as recommended by
```
This will be the same as the jsdoc above, with the exception that ``(`boolean`, default: `true`)`` *is* included here, right before the first `.`

wooorm reviewed (commented):
Thanks! Mostly some stylistic suggestion, otherwise looks good!

jorroll committed:
Feat: mapDepricatedTableCellAttrsToInlineCss opt
Adds a new plugin option, `mapDepricatedTableCellAttrsToInlineCss`, which controls whether table-related attributes are mapped to inline CSS styles. Defaults to `true`.

wooorm commented on readme.md:
```diff
@@ -174,6 +174,13 @@ React key prefix (`string`, default: `'h-'`).
+###### `options.fixTableCellAlign`
+
+Fix obsolete align attributes on table cells by turning them
+into inline styles.
+Keep it on when working with markdown, turn it off when working
+with markup for emails (`boolean`, default: `true`).
```
```suggestion
into inline styles (`boolean`, default: `true`).
Keep it on when working with markdown, turn it off when working
with markup for emails.
```

Default on the first sentence

wooorm committed:
Update lib/index.js

Signed-off-by: Titus <tituswormer@gmail.com>

wooorm committed:
Update readme.md

Signed-off-by: Titus <tituswormer@gmail.com>

wooorm commented:
Thanks, released in 7.2.0!
